pkg/apis/servicecatalog/types.go

```diff
@@ -266,10 +266,16 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// Parameters is set of properties to be
 	// passed to the underlying broker.
 	Parameters *runtime.RawExtension
 
+	// Optional: AlphaSecretParameters is reference to the secret object containing
+	// sensitive information to pass to the underlying broker. This may be
+	// empty if no secret object is specified.
+	// +optional
+	AlphaSecretParameters *v1.LocalObjectReference
```

Comment:
I think this needs to be an array, and it's more than the reference to the object.  It needs to be a reference to a key, and also have the name of the parameter.  How about:

```go
type SecretParameter struct {
  v1.SecretKeyRef
  ParameterName string
}
```

I'm not sure that SecretKeyRef from the v1 API will be usable, though, because I don't think we want the optional bit.  So, that yields:

```go
type SecretParameter struct {
  LocalObjectReference
  Key string
  ParameterName string
}
```

Thoughts?